### PR TITLE
enhancement/15892-idicator-order-in-series 

### PR DESCRIPTION
**Pull request by karolkolodziej**
Allow declaring an indicator series before the main series. See #15892.

### CODE REVIEW

File: ts/Stock/Indicators/SMA/SMAIndicator.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * */
 
 import type AxisType from '../../../Core/Axis/AxisType';
-import type Chart from '../../../Core/Chart/Chart';
+import Chart from '../../../Core/Chart/Chart.js';
```
Comment (pawelfus) [inline ts/Stock/Indicators/SMA/SMAIndicator.ts:18]: Could you move this away from type imports? Simply move them to the other imports

File: ts/Stock/Indicators/VBP/VBPIndicator.ts [hunk 2]
```diff
@@ -218,11 +219,21 @@ class VBPIndicator extends SMAIndicator {
 
         H.seriesTypes.sma.prototype.init.apply(indicator, arguments);
 
-        params = (indicator.options.params as any);
-        baseSeries = indicator.linkedParent;
-        volumeSeries = (chart.get((params.volumeSeriesID as any)) as any);
+        // Only after series are linked add some additional logic/properties.
+        addEvent(StockChart, 'afterLinkSeries', function (): void {
```
Comment (pawelfus) [inline ts/Stock/Indicators/VBP/VBPIndicator.ts:223]: Missing unbinder

File: ts/Stock/Indicators/Supertrend/SupertrendIndicator.ts [hunk 3]
```diff
@@ -190,17 +192,27 @@ class SupertrendIndicator extends SMAIndicator {
...
-        // reduced by period due to points comparison in drawGraph method
-        // (#9787)
-        options.cropThreshold = (
-            (parentOptions.cropThreshold as any) -
-            ((options.params as any).period - 1)
-        );
+        const indicator = this;
+
+        // Only after series are linked add some additional logic/properties.
+        addEvent(StockChart, 'afterLinkSeries', function (): void {
```
Comment (pawelfus) [inline ts/Stock/Indicators/Supertrend/SupertrendIndicator.ts:198]: Unbind missing

**Review by pawelfus — CHANGES_REQUESTED** [review]

**Review by pawelfus — APPROVED** [review]
Looks good, thanks!

**Review by raf18seb — APPROVED** [review]

**Review by pawellysy — APPROVED** [review]
nothing to add 👍

**Pull request merged**